test/integration/school_signup_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+class SchoolSignupTest < ActionDispatch::IntegrationTest
+
+  test "invalid school signup information" do
+    get school_registration_path
+    assert_no_difference 'School.count' do
+      post school_registration_path, school: { name:  "",
+                                               email: "user@invalid",
+                                               password:              "foo",
+                                               password_confirmation: "bar" }
+    end
+    assert_template  'schools/registrations/new'
+    assert_select 'div#error_explanation' do |element|
+      assert_select "li", 6
+      assert_match "Password is too short (minimum is 8 characters)", element.to_s
+      assert_match "Password confirmation doesn't match Password", element.to_s
+      assert_match "Email is invalid", element.to_s
+      assert_match "Name can't be blank", element.to_s
+      assert_match "Address can't be blank", element.to_s
+      assert_match "Plz can't be blank", element.to_s
```

Comment:
This looks more like an acceptance/feature test.

I would start using rspec and capybara so that if a registration succeeds, you can check that the school has all of the correct values after filling out the form with capybara and providing values there.

Then, you can do things like `expect(page).to have_content school.email`

For this test you would have something like`expect(page).to have_content "Password is too short (minimum is 8 characters)"`